src/model/consent/consent.ts

```diff
@@ -42,45 +42,62 @@
  * of 'createdAt' field may need to be tested in the future.
  */
 
-import { NotFoundError, RevokedConsentModificationError } from '../errors'
-import Knex from 'knex'
+import { NotFoundError } from '../errors'
+import Knex from 'knex';
 
 /*
  * Interface for Consent resource type
  */
 export interface Consent {
   id: string;
-  initiatorId?: string;
-  participantId?: string;
+  // participant DFSP that requested a Consent resource be made
+  participantId: string;
+  // status of the consent
   status: string;
-  credentialId?: string;
-  credentialType?: string;
-  credentialStatus?: string;
-  credentialPayload?: string;
-  credentialChallenge?: string;
+  // NOTE: not sure what purpose credentialId serves.
+  credentialId: string;
+  // credential type - currently trying to support FIDO/Generic credentials
+  credentialType: 'FIDO' | 'GENERIC' | undefined;
+  // NOTE: unsure this field is needed since `auth-service` verifies
+  //       credential on receiving them
+  credentialStatus: string;
```

Comment:
We should hold onto this, at least for auditing purposes.

In the API it can be `PENDING`, `VERIFIED` or `REVOKED`, but yes, in the database it should only really be `VERIFIED` or `REVOKED`

Reply:
Ah I see. It's the consent.status that can be revoked, not the consent.credential status (ie. the entire consent is what is revoked, not just the credential of that consent)

So in that case, I think we can remove this field.

Reply:
:+1: